app/_src/gateway/production/logging/debug-request.md

```diff
@@ -0,0 +1,175 @@
+---
+title: Debug requests
+content_type: explanation
+---
+
+{{site.base_gateway}} admins can debug requests by collecting timing information about a given request, on demand. 
+Request debugging is triggered using a secure token and the resulting data is returned in a response header named `X-Kong-Request-Debug-Output`. 
+
+Request debugging provides the following insights:
+* Time spent in various {{site.base_gateway}} components, such as plugins, DNS resolution, and load balancing. 
+* Contextual information, such as the domain name tried during these processes.
+
+{:.note}
+> **Note:** This feature is meant for live debugging. The JSON schema of the header containing the timing should never be considered static and is always subject to change.
+
+## Enable request debugging
+
+Request debugging is enabled by default and has the following configurations in [`kong.conf`](/gateway/{{page.kong_version}}/reference/configuration/):
+
+```
+request_debug = on | off # enable or disable request debugging
+request_debug_token <token> # Set debug token explicitly. Otherwise, it will be generated randomly when Kong starts, restarts, and reloads. 
+```
+
+The debug token (`X-Kong-Request-Debug-Token`) is used to authenticate the client and send the debug request, preventing abuse. 
+
+You can find the debug token in the following locations:
+* **{{site.base_gateway}} error log:** The debug token is logged in the error log (notice level) when {{site.base_gateway}} starts, restarts, or reloads. The log line will have the `[request-debug]` prefix to aid in searching.
+* **Filesystem:** The debug token is also stored in a file at `{prefix}/.request_debug_token` and updated when {{site.base_gateway}} starts, restarts, or reloads.
+
+## Debug request configuration
+
+To debug a request, add the following request headers:
+* At a minimum, you should set the `X-Kong-Request-Debug` header. 
+* If the requests originate from loopback addresses, the `X-Kong-Request-Debug-Token` header also needs to be set.
+
+### X-Kong-Request-Debug header
+
+If the `X-Kong-Request-Debug` header is set to `*`, timing information will be collected and exported for the current request.
+
+In {{site.ee_product_name}}, you can also specify a list of filters, separated by commas, to filter the scope of the time information that is collected. Each filters specifies which phase to collect timing information from. The following filters are supported:
```

Comment:
> In {{site.ee_product_name}}

Are filters an enterprise-specific feature?

Reply:
@lena-larionova Yep, from what the Google doc said:

"(EE only) You can also specify a list of filters, separated by commas, to filter the scope of the time information that is collected. The following filters are supported:"